applications/OptimizationApplication/python_scripts/solvers/helmholtz_solver.py

```diff
@@ -0,0 +1,235 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+import KratosMultiphysics.python_linear_solver_factory as linear_solver_factory
+from KratosMultiphysics.python_solver import PythonSolver
+from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import ContainerVariableDataHolderUnion
+from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import ConvertContainers
+
+class HelmholtzSolver(PythonSolver):
+    @classmethod
+    def GetDefaultParameters(cls):
+        return Kratos.Parameters("""{
+            "model_part_name"          : "",
+            "reference_model_part_name": "",
+            "echo_level"               : 0,
+            "linear_solver_settings"   : {}
+        }""")
+
+    def __init__(self, model: Kratos.Model, settings: Kratos.Parameters):
+        super().__init__(model, settings)
+
+        model_part_name = self.settings["model_part_name"].GetString()
+        reference_model_part_name = self.settings["reference_model_part_name"].GetString()
+
+        if model_part_name == "":
+            raise Exception('Please provide the model part name as the "model_part_name" (string) parameter!')
+
+        self.reference_model_part = self.model[reference_model_part_name]
+        self.main_model_part = self.model.CreateModelPart(model_part_name)
+
+        self.main_model_part.ProcessInfo[Kratos.DOMAIN_SIZE] = self.reference_model_part[Kratos.DOMAIN_SIZE]
+
+    def AddVariables(self):
+        # we add variables to the reference model part
+        # then these will be copied to the helmholtz model part
+        # with the connectivity preserve modeller
+        self.reference_model_part.AddNodalSolutionStepVariable(KratosOA.HELMHOLTZ_SOURCE_DENSITY)
+        self.reference_model_part.AddNodalSolutionStepVariable(KratosOA.HELMHOLTZ_VAR_DENSITY)
```

Comment:
Name of the Kratos variable:"VAR" does not sound very descriptive. Should it mean "variance"?

Reply:
These variable names will be changed to generic names accordingly.